### PR TITLE
Specify user input in parser error

### LINKED ISSUE #145: [PE-D] worker-edit and shift-time command doesn't handle extra parameters properly.
No details provided.

If an extra parameter `j/0` is entered, the resulting message is incorrect.

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: mkeoliya/ped#4

### LINKED ISSUE #170: [PE-D] worker-assign leave error
Wrong error message when assigning another role for the same slot. Should be john is on leave

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: ernestlim8/ped#15

### PR DESCRIPTION
**Pull request by sigmund-c**
`worker-edit 1 p/12 j/invalidprefix`
used to return
`Hourly pay should be a positive number not exceeding 1000, with a maximum of 2 decimal places`
now it returns
```
The Pay provided is invalid: '12 j/invalidprefix'.
Hourly pay should be a positive number not exceeding 1000, with a maximum of 2 decimal places
```
Resolves #145 

Also assigning conflicts now specify the already existing assignment instead of the inputted assignment
Resolves #170

### CODE REVIEW

File: src/test/java/mcscheduler/logic/commands/TakeLeaveCommandTest.java [hunk 1]
```diff
@@ -170,6 +171,11 @@ public ModelStubAlreadyHasAssignment(List<Worker> workers, List<Shift> shifts) {
         public boolean hasAssignment(Assignment assignment) {
             return true;
         }
+
+        @Override
+        public Optional<Assignment> getAssignment(Assignment toGet) {
+            return Optional.of(toGet);
+        }
```
Comment (WangZijun97) [inline src/test/java/mcscheduler/logic/commands/TakeLeaveCommandTest.java]: Don't think this is necessary as `getAssignment` was added to this stub class in a recent PR. But perhaps that can be updated to either fetch or always provide `toGet` using `orElse()` to fulfil this functionality.

**Review by WangZijun97 — COMMENTED** [review]
LGTM! Just a small thing about the stub `getAssignment()` method. Also do settle the merge conflicts and this is good to go.

File: src/test/java/mcscheduler/logic/commands/TakeLeaveCommandTest.java (hunk 1 above)
Comment (sigmund-c) [reply src/test/java/mcscheduler/logic/commands/TakeLeaveCommandTest.java]: alright, fixed!

**Review by WangZijun97 — APPROVED** [review]
LGTM!

**Pull request merged**